common/src/main/java/com/github/teamfusion/summonerscrolls/mixin/TieredItemMixin.java

```diff
@@ -0,0 +1,109 @@
+package com.github.teamfusion.summonerscrolls.mixin;
+
+import com.github.teamfusion.summonerscrolls.common.entity.ISummon;
+import com.github.teamfusion.summonerscrolls.common.util.ScrollUtil;
+import net.minecraft.MethodsReturnNonnullByDefault;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.MobSpawnType;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TieredItem;
+import net.minecraft.world.item.context.UseOnContext;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.gameevent.GameEvent;
+import org.spongepowered.asm.mixin.Mixin;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Objects;
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+@Mixin(TieredItem.class)
+public abstract class TieredItemMixin extends Item {
```

Comment:
Why is this not just an override in the ScrollItem class? Putting it here risks breaking other mods and there's no feasible reason to, given that you wrote the class that this is for in the first place.

Reply:
I couldn't really do that since the scroll item is just like the enchantment book not the actual weapon spawning in the summons.

Reply:
Ok. I'd try to find a way to do this with an Inject instead though, since Overrides are dangerous. Only one override from any mod can target any given method, so any other mod that tries to override or redirect this same method will be incompatible and I feel like there's a way to do this with an inject.